src/app/modules/consultations/consultation-profile/read-respond/read-respond.component.ts

```diff
@@ -158,20 +159,42 @@ export class ReadRespondComponent implements OnInit, AfterViewChecked {
       const answers = {...this.questionnaireForm.value};
       const value = [];
       for (const item in answers) {
-        if (answers.hasOwnProperty(item)) {
+        if (answers.hasOwnProperty(item) && answers[item] !== null) {
           if (typeof(answers[item]) === 'object') {
               const keys = Object.keys(answers[item]);
               const filtered = keys.filter(function(key) {
                   return answers[item][key];
               });
               answers[item] = filtered;
+              answers[item].forEach(ele => {
+                if (ele === 'other') {
+                  value.push({
+                    question_id: item,
+                    is_other: true,
+                    other_option_answer: answers['other_answer-' + item],
+                    answer: answers[item].filter(val => {
+                      return val !== 'other';
+                    })
+                  });
+                }
+              });
           }
-          value.push({
-            question_id: item,
-            answer: answers[item]
-          });
+          if (answers[item] === 'other') {
+            value.push({
+              question_id: item,
+              is_other: true,
+              other_option_answer: answers['other_answer-' + item],
+            });
+          } else if (!(item.includes('other'))) {
```

Comment:
@preethijayanthi I think this block is still running for answers with other. I tested with Consultation 155 on staging and the response object had a duplicate- 
![image](https://user-images.githubusercontent.com/9947313/91744486-37823e80-ebd7-11ea-9f94-25cbb5975f58.png)

Reply:
Also when only the other option is selected, we do not need to send `answer: []` like here- 
![image](https://user-images.githubusercontent.com/9947313/91750586-f858eb00-ebe0-11ea-946a-22828a0af127.png)